GTG/gtk/general_preferences.py

```diff
@@ -0,0 +1,249 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Getting Things GNOME! - a personal organizer for the GNOME desktop
+# Copyright (c) 2008-2013 - Lionel Dricot & Bertrand Rousseau
+#
+# This program is free software: you can redistribute it and/or modify it under
+# the terms of the GNU General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later
+# version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program.  If not, see <http://www.gnu.org/licenses/>.
+# -----------------------------------------------------------------------------
+
+"""General Preferences represents one of the children
+in Preferences window. It is viewed in its stack.
+It enables user to set the most general settings of GTG."""
+
+import os
+
+from gi.repository import Gtk, Gdk
+
+from GTG.core.dirs import UI_DIR
+from GTG.core.translations import _
+from GTG.tools import autostart
+from GTG.tools import shortcut
+
+
+class GeneralPreferences(object):
+
+    GENERAL_PREFERENCES_UI = os.path.join(UI_DIR, "general_preferences.ui")
+    INVALID_COLOR = Gdk.Color(50000, 0, 0)
+
+    def __init__(self, req, vmanager):
+        self.req = req
+        self.config = self.req.get_config('browser')
+        builder = Gtk.Builder()
+        builder.add_from_file(self.GENERAL_PREFERENCES_UI)
+
+        self.ui_widget = builder.get_object("general_pref_window")
+        self.autostart_button = builder.get_object("autostart_button")
+        self.preview_button = builder.get_object("preview_button")
+        self.bg_color_button = builder.get_object("bg_color_button")
+        self.shortcut_button = builder.get_object("shortcut_button")
+        self.font_button = builder.get_object("font_button")
+        self.shortcut_popover = builder.get_object("shortcut_popover")
+        self.set_shortcut = builder.get_object("set_shortcut")
+
+        self.shortcut = ShortcutWidget(builder)
+
+        self.timer = vmanager.timer
+        self.refresh_time = builder.get_object("time_entry")
+
+        self.set_editor_font()
+
+        builder.connect_signals(self)
+
+    def get_ui(self):
+        '''
+        This method returns widget displayed in Preferences window.
+        '''
+        return self.ui_widget
+
+    def on_shortcut_popover(self, widget):
+        self.shortcut_popover.show()
+
+    def set_editor_font(self):
```

Comment:
This should be get_editor_font() (or get_default_editor_font()) and it should return the default value of editor font.  Please do the setting of the font inside of `_refresh_preferences_store` as it would make this function much easier to test.